contracts/vesting/cw20-vest/src/contract.rs

```diff
@@ -0,0 +1,1219 @@
+                )?;
+                Ok(Response::new()
+                    .add_message(unstake_msg)
+                    .add_attribute("action", "vest")
+                    .add_attribute("from", info.sender)
+                    .add_attribute("amount", claimable)
+                    .add_attribute("claim_duration", format!("{}", duration)))
+            }
+        }
+    }
+
+    pub fn claim(deps: DepsMut, env: Env, info: MessageInfo) -> Result<Response, ContractError> {
+        if !ACTIVATED.load(deps.storage)? {
+            return Err(ContractError::Unfunded {})
+        }
+
+        let release = CLAIMS.claim_tokens(deps.storage, &info.sender, &env.block, None)?;
+        if release.is_zero() {
+            return Err(ContractError::NothingToClaim {});
+        }
+        let config = CONFIG.load(deps.storage)?;
+        let balance = query_balance(deps.as_ref(), &config.token_address, &env.contract.address)?;
+        if balance < release {
+            let claim_msg = WasmMsg::Execute {
+                contract_addr: config.stake_address.to_string(),
+                msg: to_binary(&cw20_stake::msg::ExecuteMsg::Claim {})?,
+                funds: vec![],
+            };
+            let transfer_msg = cosmwasm_std::WasmMsg::Execute {
+                contract_addr: config.token_address.to_string(),
+                msg: to_binary(&cw20::Cw20ExecuteMsg::Transfer {
+                    recipient: info.sender.to_string(),
+                    amount: release,
+                })?,
+                funds: vec![],
+            };
+            Ok(Response::new()
+                .add_message(claim_msg)
+                .add_message(transfer_msg)
+                .add_attribute("action", "claim")
+                .add_attribute("from", info.sender)
+                .add_attribute("amount", release))
+        } else {
+            let transfer_msg = cosmwasm_std::WasmMsg::Execute {
+                contract_addr: config.token_address.to_string(),
+                msg: to_binary(&cw20::Cw20ExecuteMsg::Transfer {
+                    recipient: info.sender.to_string(),
+                    amount: release,
+                })?,
+                funds: vec![],
+            };
+            Ok(Response::new()
+                .add_message(transfer_msg)
+                .add_attribute("action", "claim")
+                .add_attribute("from", info.sender)
+                .add_attribute("amount", release))
+        }
+    }
+
+    pub fn update_config(
+        deps: DepsMut,
+        _env: Env,
+        info: MessageInfo,
+        new_owner: Option<String>,
+        new_manager: Option<String>
+    ) -> Result<Response, ContractError> {
+        let new_owner = new_owner
+            .map(|new_owner| deps.api.addr_validate(&new_owner))
+            .transpose()?;
+        let new_manager = new_manager
+            .map(|new_manager| deps.api.addr_validate(&new_manager))
+            .transpose()?;
```

Comment:
if you were making a DAO with this vesting contract would you like to appoint a manger, or could we remove this?